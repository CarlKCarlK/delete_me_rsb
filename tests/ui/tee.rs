use itertools::Itertools;
use range_set_int::{RangeSetInt, SortedDisjointIterator};

fn main() {
    let a = [(1, -1), (-3, 4)];
    let (a0, a1) = a.iter().tee();
    let _c = a0.union(a1);
}
