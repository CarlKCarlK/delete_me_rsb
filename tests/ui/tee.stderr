warning: unused import: `RangeSetInt`
 --> tests/ui/tee.rs:2:21
  |
2 | use range_set_int::{RangeSetInt, SortedDisjointIterator};
  |                     ^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0599]: the method `bitor` exists for struct `Tee<std::slice::Iter<'_, ({integer}, {integer})>>`, but its trait bounds were not satisfied
 --> tests/ui/tee.rs:7:17
  |
7 |     let _c = a0.bitor(a1);
  |                 ^^^^^ method cannot be called on `Tee<std::slice::Iter<'_, ({integer}, {integer})>>` due to unsatisfied trait bounds
  |
 ::: $CARGO/itertools-0.10.5/src/tee.rs
  |
  | pub struct Tee<I>
  | -----------------
  | |
  | doesn't satisfy `<_ as Iterator>::Item = RangeInclusive<_>`
  | doesn't satisfy `_: SortedDisjointIterator<_>`
  | doesn't satisfy `_: range_set_int::sorted_disjoint_iter::SortedDisjoint`
  |
  = note: the following trait bounds were not satisfied:
          `<Tee<std::slice::Iter<'_, ({integer}, {integer})>> as Iterator>::Item = RangeInclusive<_>`
          which is required by `Tee<std::slice::Iter<'_, ({integer}, {integer})>>: SortedDisjointIterator<_>`
          `Tee<std::slice::Iter<'_, ({integer}, {integer})>>: range_set_int::sorted_disjoint_iter::SortedDisjoint`
          which is required by `Tee<std::slice::Iter<'_, ({integer}, {integer})>>: SortedDisjointIterator<_>`
          `<&Tee<std::slice::Iter<'_, ({integer}, {integer})>> as Iterator>::Item = RangeInclusive<_>`
          which is required by `&Tee<std::slice::Iter<'_, ({integer}, {integer})>>: SortedDisjointIterator<_>`
          `&Tee<std::slice::Iter<'_, ({integer}, {integer})>>: Iterator`
          which is required by `&Tee<std::slice::Iter<'_, ({integer}, {integer})>>: SortedDisjointIterator<_>`
          `&Tee<std::slice::Iter<'_, ({integer}, {integer})>>: range_set_int::sorted_disjoint_iter::SortedDisjoint`
          which is required by `&Tee<std::slice::Iter<'_, ({integer}, {integer})>>: SortedDisjointIterator<_>`
          `<&mut Tee<std::slice::Iter<'_, ({integer}, {integer})>> as Iterator>::Item = RangeInclusive<_>`
          which is required by `&mut Tee<std::slice::Iter<'_, ({integer}, {integer})>>: SortedDisjointIterator<_>`
          `&mut Tee<std::slice::Iter<'_, ({integer}, {integer})>>: range_set_int::sorted_disjoint_iter::SortedDisjoint`
          which is required by `&mut Tee<std::slice::Iter<'_, ({integer}, {integer})>>: SortedDisjointIterator<_>`

warning: unused import: `SortedDisjointIterator`
 --> tests/ui/tee.rs:2:34
  |
2 | use range_set_int::{RangeSetInt, SortedDisjointIterator};
  |                                  ^^^^^^^^^^^^^^^^^^^^^^
